### PR TITLE
feat(user-dockerfile): Add user argument to DockerConfiguration

### PR DESCRIPTION
**Pull request by cjidboon94**
## Describe changes
I implemented a `user` argument to achieve being able to set the user in the generated Dockerfile via the docker USER command.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/mlops-stacks/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: tests/unit/utils/test_pipeline_docker_image_builder.py [hunk 1]
```diff
@@ -38,6 +38,28 @@ def test_including_active_profile_in_build_context(tmp_path: Path):
     assert not config_path.exists()
 
 
+def test_check_user_is_set(tmp_path: Path):
```
Comment (schustmi) [inline tests/unit/utils/test_pipeline_docker_image_builder.py:41]:
```suggestion
def test_check_user_is_set():
```
Don't think the `tmp_path` is needed here

**Review by schustmi — CHANGES_REQUESTED** [review]
One tiny nit, otherwise looks great!

**Comment by schustmi** [comment]
Also, the target branch of this PRs needs to be `develop` instead of `main`

File: tests/unit/utils/test_pipeline_docker_image_builder.py (hunk 1 above)
Comment (cjidboon94) [reply tests/unit/utils/test_pipeline_docker_image_builder.py]: Agreed. Removed

**Comment by schustmi** [comment]
@cjidboon94 The linting is still failing, running the `scripts/lint.sh` script from the git repo root directory should hopefully fix that

File: tests/unit/utils/test_pipeline_docker_image_builder.py [hunk 2]
```diff
@@ -38,6 +38,29 @@ def test_including_active_profile_in_build_context(tmp_path: Path):
...
+
+    config = DockerConfiguration(
+        install_stack_requirements=False,
+        requirements=None,
+        required_integrations=[],
+        user="test_user",
+        replicate_local_python_environment="pip_freeze",
+    )
+    generated_dockerfile = PipelineDockerImageBuilder._generate_zenml_pipeline_dockerfile("test:test", config)
+    print(generated_dockerfile)
```
Comment (htahir1) [inline tests/unit/utils/test_pipeline_docker_image_builder.py:60]: Not sure if the print here is neccessary?

File: tests/unit/utils/test_pipeline_docker_image_builder.py [hunk 3]
```diff
@@ -38,6 +38,29 @@ def test_including_active_profile_in_build_context(tmp_path: Path):
     assert not config_path.exists()
 
 
+def test_check_user_is_set():
```
Comment (htahir1) [inline tests/unit/utils/test_pipeline_docker_image_builder.py:41]: great test!

**Review by htahir1 — COMMENTED** [review]
Added a review as well, as this caught my eye. Awesome contribution, small, but significant!

File: tests/unit/utils/test_pipeline_docker_image_builder.py (hunk 2 above)
Comment (cjidboon94) [reply tests/unit/utils/test_pipeline_docker_image_builder.py]: Agreed. Removed

**Review by htahir1 — CHANGES_REQUESTED** [review]
@cjidboon94 Awesome, now looks clean! Just one small change needed. The formatting is still failing, running the `scripts/format.sh` script from the git repo root directory should hopefully fix that

**Review by htahir1 — APPROVED** [review]
Awesome, LGTM 🚀 ! Thank you for the awesome contribution @cjidboon94!

**Pull request merged**